src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/CohereEmbeddingClient.java

```diff
@@ -74,8 +94,19 @@ public Uni<List<float[]>> vectorize(
     EmbeddingRequest request =
         new EmbeddingRequest(texts.toArray(textArray), modelName, input_type);
     Uni<EmbeddingResponse> response =
-        embeddingProvider.embed(
-            "Bearer " + (apiKeyOverride.isPresent() ? apiKeyOverride.get() : apiKey), request);
+        embeddingProvider
+            .embed(
+                "Bearer " + (apiKeyOverride.isPresent() ? apiKeyOverride.get() : apiKey), request)
+            .onFailure(
+                throwable -> {
+                  return (throwable.getCause() != null
+                      && throwable.getCause() instanceof JsonApiException jae
+                      && jae.getErrorCode() == ErrorCode.EMBEDDING_PROVIDER_TIMEOUT);
+                })
+            .retry()
+            .withBackOff(Duration.ofMillis(requestProperties.retryDelayInMillis()))
+            .atMost(requestProperties.maxRetries());
```

Comment:
Does this take in max number retries, or total number of calls? (differ by 1)

Reply:
Documentation says `You can configure the number of retries using atMost`, so I believe it is number of retries attempts.